var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -51,10 +51,13 @@ class Julia(MakefilePackage):
     # Note, we just use link_llvm_dylib so that we not only get a libLLVM,
     # but also so that llvm-config --libfiles gives only the dylib. Without
     # it it also gives static libraries, and breaks Julia's build.
+    # Also disable a few unneeded variants since they can cause
+    # https://github.com/spack/spack/issues/36954
     depends_on(
         "llvm"
         " targets=amdgpu,bpf,nvptx,webassembly"
         " version_suffix=jl +link_llvm_dylib libunwind=none"
+        " ~clang ~gold ~polly"
```

Comment:
I can live with this, but let's reassess when somebody wants to use clang for some reason :p

Reply:
This doesn't affect `clang` as a compiler though, right?

Reply:
Although I'd rather have the underlying issue fixed... I don't think `~clang ~polly` are related to the issue?